guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/preferences/GAVPreferences.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.common.services.project.preferences;
+
+import org.uberfire.ext.preferences.shared.PropertyFormType;
+import org.uberfire.ext.preferences.shared.annotations.Property;
+import org.uberfire.ext.preferences.shared.annotations.WorkbenchPreference;
+import org.uberfire.ext.preferences.shared.bean.BasePreference;
+
+@WorkbenchPreference(identifier = "GAVPreferences",
+        bundleKey = "GAVPreferences.Label",
+        parents = "GeneralPreferences")
+public class GAVPreferences implements BasePreference<GAVPreferences> {
+
+    @Property(bundleKey = "GAVPreferences.ConflictingGAVCheckDisabled.Label",
+            formType = PropertyFormType.BOOLEAN)
+    private boolean conflictingGAVCheckDisabled;
+
+    @Property(bundleKey = "GAVPreferences.ChildGAVEditEnabled.Label",
+            formType = PropertyFormType.BOOLEAN)
+    private boolean childGAVEditEnabled;
+
+    @Override
+    public GAVPreferences defaultValue( final GAVPreferences defaultValue ) {
+        final String conflictingGAVCheckDisabledSystemProperty = System.getProperty( "org.guvnor.project.gav.check.disabled", "false" );
```

Comment:
This class is in "-api" and covered by the GWT module's ```<source>``` paths; so will be available client-side as well as server-side. Therefore the ```System.getProperty(...)``` calls are likely to fail on the client. The old way we made System Properties available client-side was with ```GAVPreferencesLoader``` which ran on the server and ultimately lead to the initialisation of  ```ApplicationPreferences``` on the client (see [here](https://github.com/droolsjbpm/kie-wb-common/blob/master/kie-wb-common-screens/kie-wb-common-workbench/kie-wb-common-workbench-client/src/main/java/org/kie/workbench/common/workbench/client/entrypoint/DefaultWorkbenchEntryPoint.java#L82)). Can you please advise how the new mechanism works?